### ISSUE TITLE
compilation error with python 3.7 / gcc 8.1.1 on fedora 29

### ISSUE THREAD

**Issue by totaam** [inducer/pycuda#184]
```
gcc -pthread -Wno-unused-result -Wsign-compare -DDYNAMIC_ANNOTATIONS_ENABLED=1 -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D_GNU_SOURCE -fPIC -fwrapv -O3 -DNDEBUG -fPIC -DBOOST_ALL_NO_LIB=1 -DBOOST_THREAD_BUILD_DLL=1 -DBOOST_MULTI_INDEX_DISABLE_SERIALIZATION=1 -DBOOST_PYTHON_SOURCE=1 -Dboost=pycudaboost -DBOOST_THREAD_DONT_USE_CHRONO=1 -DPYGPU_PACKAGE=pycuda -DPYGPU_PYCUDA=1 -DHAVE_GL=1 -Isrc/cpp -Ibpl-subset/bpl_subset -I/usr/local/cuda/include -I/usr/lib64/python3.7/site-packages/numpy/core/include -I/usr/lib64/python3.7/site-packages/numpy/core/include -I/usr/lib64/python3.7/site-packages/numpy/core/include -I/usr/include/python3.7m -c bpl-subset/bpl_subset/libs/python/src/converter/builtin_converters.cpp -o build/temp.linux-x86_64-3.7/bpl-subset/bpl_subset/libs/python/src/converter/builtin_converters.o
In file included from bpl-subset/bpl_subset/boost/python/handle.hpp:10,
                 from bpl-subset/bpl_subset/libs/python/src/converter/builtin_converters.cpp:6:
bpl-subset/bpl_subset/boost/python/cast.hpp: In function �void pycudaboost::python::detail::assert_castable(pycudaboost::type<Target>*)�:
bpl-subset/bpl_subset/boost/python/cast.hpp:73:20: warning: typedef �must_be_a_complete_type� locally defined but not used [-Wunused-local-typedefs]
       typedef char must_be_a_complete_type[sizeof(T)];
                    ^~~~~~~~~~~~~~~~~~~~~~~
In file included from bpl-subset/bpl_subset/boost/bind/mem_fn.hpp:25,
                 from bpl-subset/bpl_subset/boost/mem_fn.hpp:22,
                 from bpl-subset/bpl_subset/boost/function/detail/prologue.hpp:18,
                 from bpl-subset/bpl_subset/boost/function/function_template.hpp:13,
                 from bpl-subset/bpl_subset/boost/function/detail/maybe_include.hpp:13,
                 from bpl-subset/bpl_subset/boost/function/function0.hpp:11,
                 from bpl-subset/bpl_subset/boost/python/errors.hpp:13,
                 from bpl-subset/bpl_subset/boost/python/handle.hpp:11,
                 from bpl-subset/bpl_subset/libs/python/src/converter/builtin_converters.cpp:6:
bpl-subset/bpl_subset/boost/get_pointer.hpp: At global scope:
bpl-subset/bpl_subset/boost/get_pointer.hpp:25:40: warning: �template<class> class std::auto_ptr� is deprecated [-Wdeprecated-declarations]
 template<class T> T * get_pointer(std::auto_ptr<T> const& p)
                                        ^~~~~~~~
In file included from /usr/include/c++/8/memory:80,
                 from bpl-subset/bpl_subset/boost/function/function_base.hpp:16,
                 from bpl-subset/bpl_subset/boost/function/detail/prologue.hpp:17,
                 from bpl-subset/bpl_subset/boost/function/function_template.hpp:13,
                 from bpl-subset/bpl_subset/boost/function/detail/maybe_include.hpp:13,
                 from bpl-subset/bpl_subset/boost/function/function0.hpp:11,
                 from bpl-subset/bpl_subset/boost/python/errors.hpp:13,
                 from bpl-subset/bpl_subset/boost/python/handle.hpp:11,
                 from bpl-subset/bpl_subset/libs/python/src/converter/builtin_converters.cpp:6:
/usr/include/c++/8/bits/unique_ptr.h:53:28: note: declared here
   template<typename> class auto_ptr;
                            ^~~~~~~~
bpl-subset/bpl_subset/libs/python/src/converter/builtin_converters.cpp: In function �void* pycudaboost::python::converter::{anonymous}::convert_to_cstring(PyObject*)�:
bpl-subset/bpl_subset/libs/python/src/converter/builtin_converters.cpp:51:35: error: invalid conversion from �const void*� to �void*� [-fpermissive]
       return PyUnicode_Check(obj) ? _PyUnicode_AsString(obj) : 0;
error: command 'gcc' failed with exit status 1
error: Bad exit status from /var/tmp/rpm-tmp.8Xa4FI (%build)
    Bad exit status from /var/tmp/rpm-tmp.8Xa4FI (%build)
```

**Comment by inducer** [comment]
Two ways out of this IMO:

* Upgrade the boost subset in PyCUDA
* Retool PyCUDA to use pybind11 (much less work than it seems--because the two are largely the same library)

I'm leaning towards the latter, and I'd be happy to have help.

**Comment by inducer** [comment]
https://github.com/inducer/bpl-subset/pull/2, by way of f4458dd26fecffdff6d701c9780ce258380e87b8, may help here. I would be grateful if you could test this out and report if it works for you. If so, I will roll a new release.

**Comment by inducer** [comment]
You likely also want a30b867 for 3.7 compatibility.

**Comment by totaam** [comment]
@inducer confirmed that current head builds fine now on Fedora 29.

**Comment by inducer** [comment]
Thanks! I've just released 2018.1.1 to address this. Thanks again for reporting this.
